django/db/backends/oracle/introspection.py

```diff
@@ -48,6 +48,26 @@ def get_field_type(self, data_type, description):
 
         return super().get_field_type(data_type, description)
 
+    def only_nonempty_tables(self, tables):
+        """
+        Take a list of tables, return a list of tables that have at least
+        one row.
+        """
+        if not tables:
+            return []
+        sql_list = []
+        for table in tables:
+            sql = """( SELECT '{}' AS has_rows FROM {} {} )""".format(
+                table, self.connection.ops.quote_name(table),
+                self.connection.ops.limit_offset_sql(0, 1)
```

Comment:
Looks like we could use this in the base class already and avoid the need for a special case in Oracle?